### ISSUE TITLE
Voting - Official deactivation of the push notification functionality

### ISSUE THREAD

**Issue by rafaelsetragni** [rafaelsetragni/awesome_notifications#97]
Hello everyone,

When I decided to create this plugin, there was still no way to run Dart code while the application was terminated/ killed. Therefore, to make use of all available resources of the plugin, even when the application was finished, i had to create this plugin with local notifications and push notifications, to be able to natively create the notification while the app was terminated.

Since then, two situations have occurred that made me rethink the need for the AwesomeNotifications plugin to work with push messages:

1 - It is now possible to run Dart codes while the application is closed, using the firebase `onBackgroundMessage` method. This also eliminates the need to use **iOS Extensions** and eliminates many of the problems that our developers have faced on the apple platform.

2 - Many of the developers are using the `firebase_messaging` and `awesome_notifications` plugins concurrently, and both of then are pushing notifications at same time. This creates several problems, due to the two plugins accessing the push feature, which is a global feature of the application.

Therefore, I am suggesting to permanently remove the push notification functionality from the AwesomeNotifications plugin and leave this role to the firebase_messaging plugin.

I would like your opinion if I should remove this feature or not and leave this plugin only for local notifications.

To vote, please access the link bellow:
[ SurveyMonkey - Voting - Official deactivation of the push notification functionality](https://pt.surveymonkey.com/r/7QJ5K9J)

**Comment by efraespada** [comment]
I previously used `firebase_messaging` and `flutter_local_notifications` before starting with `awesome_notifications`. I don't realize yet which different implications has this decision.

But this is my point of view:

With `firebase_messaging` and `flutter_local_notifications` (what I previously used) I had to configure `firebase_messaging`, test multiple bodies (of push notifications) for both OS (Android and iOS), handle the notifications in different scenarios and run whatever with `flutter_local_notifications`. 👎🏼 

Now with `awesome_notifications`, I don't care about `firebase_messaging`, how to handler the incoming notifications, or the push notifications' body. 👍🏼 

---- 

> using the firebase onBackgroundMessage method. This also eliminates the need to use iOS Extensions 

Now I can use ringtones for notification calls, display images in the notifications.. thanks to the `AwesomeContentExtension`, that's right? Do we still have the same display options without those **iOS Extensions**?

----

> Many of the developers are using the firebase_messaging and awesome_notifications plugins concurrently

What sense has this? I'm currently developing a messaging application and I don't need to set up or work with `firebase_messaging` anymore (at least directly I mean).

---- 

Everything is ordered and working fine 😄 

So.. why change?

**Comment by rafaelsetragni** [comment]
> ----
> 
> 
> 
> > Many of the developers are using the firebase_messaging and awesome_notifications plugins concurrently
> 
> 
> 
> What sense has this? I'm currently developing a messaging application and I don't need to set up or work with `firebase_messaging` anymore (at least directly I mean).
> 

That was what I thought, but in practice many developers discovered this plugin after had used and implemented many functions based on firebase_messaging with flutter_local_notification.

So, thinking in not lose job already done, they keep firebase_messaging with awesome_notifications, and both of then try concurrently access the notification data. This creates many undesired issues and time to figure out what's going on.

**Comment by ralmarri12** [comment]
Hello @rafaelsetragni , I believe what you did is really great as all developers believes that too. What I think makes people confused, is that the way this plugin integrate with FCM. I for example, followed the instructions to install FCM, like creating Application.kt, write those lines in build.gradle .. etc. the question makes me and other confused, should we remove all things related to firebase_messaging? as I believe also your plugin will help me do what I need more than the firebase_messaging.

what I mean, its just people who come from firebase_messaging messaging there is no integration instructions help 'em in the read me file.

so my suggestions is, keep it up just help us with the README file.

Thank you Rafael.

**Comment by whatamelon** [comment]
Agree with @ralmarri12. 
I think people confused for immigrate firebase_messaging official library to awosome_notification.
It was a problem for me before.

So, In library's docs can provide these steps for immigrate, it would be very helpful to many people.

Also, one of awesome_notification's user, I can help to write immigrate docs with my pleasure.

Thx to @rafaelsetragni  for make good library :D

**Comment by MaheshPeri19** [comment]
> > > Many of the developers are using the firebase_messaging and awesome_notifications plugins concurrently
> > 
> > 
> > What sense has this? I'm currently developing a messaging application and I don't need to set up or work with `firebase_messaging` anymore (at least directly I mean).
> 
> That was what I thought. But in practice many developers discovered this plugin after had used and implemented many functions based on firebase_messaging with flutter_local_notification.
> 
> So, thinking in not lose job already done, they keep firebase_messaging with awesome_notifications, and both of then try concurrently access the notification data. This creates many undesired issues and time to figure out what's going on.

Yes. Exactly. I (and many developers) used firebase_messaging and flutter_local_notifications before starting with awesome_notifications. I required action buttons, so that started to use awesome_notifications.

**Comment by GioPan04** [comment]
> Hello everyone,
> 
> When I decided to create this plugin, there was still no way to run Dart code while the application was terminated/ killed. Therefore, to make use of all available resources of the plugin, even when the application was finished, i had to create this plugin with local notifications and push notifications, to be able to natively create the notification while the app was terminated.
> 
> Since then, two situations have occurred that made me rethink the need for the AwesomeNotifications plugin to work with push messages:
> 
> 1 - It is now possible to run Dart codes while the application is closed, using the firebase `onBackgroundMessage` method. This also eliminates the need to use **iOS Extensions** and eliminates many of the problems that our developers have faced on the apple platform.
> 
> 2 - Many of the developers are using the `firebase_messaging` and `awesome_notifications` plugins concurrently, and both of then are pushing notifications at same time. This creates several problems, due to the two plugins accessing the push feature, which is a global feature of the application.
> 
> Therefore, I am suggesting to permanently remove the push notification functionality from the AwesomeNotifications plugin and leave this role to the firebase_messaging plugin.
> 
> I would like your opinion if I should remove this feature or not and leave this plugin only for local notifications.
> 
> To vote, please access the link bellow:
> [ SurveyMonkey - Voting - Official deactivation of the push notification functionality](https://pt.surveymonkey.com/r/7QJ5K9J)

I just discovered this plugin and it's awesome! Before I had to write my own implementation of notifications that I need when a new fcm notification was recived, but I don't know how to write iOS implementation, and this plugin does all of that I need. Please don't remove the fcm functionality, it's so handy.

Btw, why don't create a callback or something that replace the official plugin to the end user? So the end user don't need to have firebase_messaging.

**Comment by rafaelsetragni** [comment]
The survey reach the free limit. I gonna post the results soon.

**Comment by rafaelsetragni** [comment]
![image](https://user-images.githubusercontent.com/40064496/107036811-f5b67f00-6798-11eb-82fe-75b4401a5bcb.jpeg)

**Comment by GioPan04** [comment]
> ![image](https://user-images.githubusercontent.com/40064496/107036811-f5b67f00-6798-11eb-82fe-75b4401a5bcb.jpeg)

Why remove it? You can't make a setting to enable it? It's so useful the connection with fcm

**Comment by rafaelsetragni** [comment]
As you can see, the decision **was very tight!**

I saw and avaliated all the interactions on voting about both sides, and took every single one in consideration, specially with the comments.

Many people enjoyed the fact that this plugin already do both things and makes everything more easy. This helped many initial developers to implement the notifications localy and for push.

But, many other developers preffer to remove the push functionality, not only because they already used `firebase_messaging` plugin, but because they understood that those other projects that only use `awesome_notification` can be easily modified to use firebase apart.

And, for last, remove the push functionality will eliminate a lot of problems that we are facing, such as:

* iOS content extensions
 * pod files issues in release mode
* Issues with 3rd projects that are using firebase_messaging with awesome_notification, untill they realise that there is a problem.
* Problems with native libraries out of sync against they last version, resulting in a double global effort from flutter community.

So, **I will remove the push functionality**, but also I will leave a special notice in the README file about this decision, in addition to maintaining **the push topics** with all instructions for integrating with the `firebase_messaging` plugin (and even others later on).

I apologize to everyone who believed in this library for the inconvenience that you will have to go through to adapt to the new structure. As this plugin is still in development, I need to take advantage of these learnings to modify when necessary and make a better final product, which serves the entire community, giving more freedom of creation and more power in the hands of the developers.

I promisse to make my best to do this transition smooth as possible to everyone, and also i belive this is the best solution for the future.

**Thank you so much for all the support** and i hope soon will have a better plugin to help every one.

**Comment by GioPan04** [comment]
@rafaelsetragni there will be a easy way to (re)implement the push notifications feature?
For example the plugin can expose a static method that automatically parse the data coming from the official fcm plugin and shows a notification.
What I mean is if the plugin will be ready for the connection, but will be the final developer to "connect" the callbacks from fcm to the method of this plugin

**Comment by rafaelsetragni** [comment]
> What I mean is if the plugin will be ready for the connection, but will be the final developer to "connect" the callbacks from fcm to the method of this plugin

Exactly! And i will leave all instructions to make it easy to do.
And also, you gonna end with more development freedom to handle the push notification.

**Comment by MaheshPeri19** [comment]
@rafaelsetragni 

When can we expect new updated package in live ?

**Comment by rafaelsetragni** [comment]
Maybe in next week. I need to compile all current demands plus the push deactivation itself.

**Comment by KonstantinosD85** [comment]
To be honest i started using `awesome_notification` without installing `firebase_messaging`. Until i realized i had to subscribe and unsubscribe my devices to topics. So i installed both without any issue until now. But if you think it is a cleaner solution i am with you. Luckily i am not in production yet. Thanks for this awesome plugin.

**Comment by MaheshPeri19** [comment]
> Maybe in next week. I need to compile all current demands plus the push deactivation itself.

@rafaelsetragni 

Any update on this ?

**Comment by GioPan04** [comment]
I have an app in production and I would like to use this plugin... There are any updates?

**Comment by rafaelsetragni** [comment]
I gonna release soon a final .5 version, with push notifications yet activated. After that, i will remove definetely the push feature and release a .6 version

**Comment by MaheshPeri19** [comment]
@rafaelsetragni 

Any update on this ? We are waiting for this feature to remove push notifications and use only for local notifications.

And also, will you work these changes with Flutter 1.x.x or 2.0 new version ?

I think many developers will not migrate immediately to flutter 2.0 version.

**Comment by MaheshPeri19** [comment]
@rafaelsetragni 

In version 0.0.5+8, you have mentioned "FCM service with comments removed for iOS." 
Which means completely removed FCM from this package both for Android and iOS ? Or still need to do some changes to complete that functionality ?

**Comment by rafaelsetragni** [comment]
No, it's still working. I only removed the comment tag from the FCM classes, which I used to test whether the fcm could be removed or not, but I forgot to delete it later.

FCM will be completely removed in version 0.0.6.

**Comment by MaheshPeri19** [comment]
Okay. Thank you. Please release v 0.0.6 very soon. We are waiting for this. Then only we can submit our app to production.

**Comment by rafaelsetragni** [comment]
This plugin is still in development. Only plug-ins above 1.0.0 are considered stable.

If you intend to use it in production, check if it is working correctly on Android 10 or higher, as I was not able to run the emulator on my local machine. I am receiving BSOD all the time.

**Comment by MaheshPeri19** [comment]
Okay. Sure.

**Comment by scalz** [comment]
Hello,
I've been using local notif plugin, so I'm a new user :)

on my side I'm testing your plugin on : android10 real device, and also in emulator with android11. 
Looks good so far. But I'm just testing local notifications. 
I'm also using firebase messaging plugin for background, and am eager to test your next release for local notif in background msg.

I just wonder how it will work, when receiving a background msg from FCM in android, and call an awesome notif with actions, will that create two notifications in android ?? (maybe a stupid question and I should read docs..^^)

Keep the good work, your package looks really promising

**Comment by rafaelsetragni** [comment]
I gonna develop a example to show how to do that integration and also document it on our Readme file. Unfortunately, i don't start to code this part yet, so i dont know yet how to do it in details.

**Comment by MaheshPeri19** [comment]
How to completely disable this package for iOS only ?

It is working perfectly for android for me. For Android, i have checked the Platform.isAndroid, then only i intialized this package for Android. Below is the code.

```
 if (Platform.isAndroid) {
      initializeAwesomeNotifications();
    }
```
But it is initializing for iOS also and ios push notifications are not working and getting below error. 

`"receiveNotification: notification discarted"
`
I am aware of opened issue regarding discart notification.

Is it anyway to disable awesome notifications for iOS completely and use it for android for now ?

Mainly i am using awesome notifications for action buttons. Because of push notificatin issues, i am not able to move forward.

**Comment by MaheshPeri19** [comment]
@rafaelsetragni 

Please help me on this to work my app without this awesome package for iOS only. 
I want to use this package for Android as it is working perfectly.  So i want to remove for only iOS. Since month, you are not updating any code for this.

**Comment by rafaelsetragni** [comment]
In fact, im updating it in a different branch, and im doing it for free, please be patient.

Now i need to test it in android and iOS. Everything ia already coded for 0.0.6 version.

**Comment by MaheshPeri19** [comment]
Okay. Thank you. 

When can we expect 0.0.6 version in live ?

**Comment by mapacheverdugo** [comment]
@rafaelsetragni what is the status of 0.0.6? I discovered this plugin a few days ago and I want to take a look and try to contribute but I don't know what branch you are working on, what has been done, or what the next steps are

**Comment by rafaelsetragni** [comment]
Right now i have two 0.0.6 versions, one made by myself and another one by the comunity, and im mergin those two versions to release the definitive 0.0.6.

The good news is the new version is already null safety. Soon i gonna release it.
